compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -104,21 +104,6 @@ class Definitions {
    *      trait ContextFunctionN[-T0,...,-T{N-1}, +R] extends Object {
    *        def apply(using $x0: T0, ..., $x{N_1}: T{N-1}): R
    *      }
-   *
-   *  ErasedFunctionN traits follow this template:
-   *
-   *      trait ErasedFunctionN[-T0,...,-T{N-1}, +R] extends Object {
-   *        def apply(erased $x0: T0, ..., $x{N_1}: T{N-1}): R
-   *      }
-   *
-   *  ErasedContextFunctionN traits follow this template:
-   *
-   *      trait ErasedContextFunctionN[-T0,...,-T{N-1}, +R] extends Object {
-   *        def apply(using erased $x0: T0, ..., $x{N_1}: T{N-1}): R
-   *      }
-   *
-   *  ErasedFunctionN and ErasedContextFunctionN erase to Function0.
-   *
    *  ImpureXYZFunctionN follow this template:
    *
    *      type ImpureXYZFunctionN[-T0,...,-T{N-1}, +R] = {*} XYZFunctionN[T0,...,T{N-1}, R]
```

Comment:
Does this work with erased parameters? 

The old encoding seemed to support `ImpureErasedFunctionN`/`ImpureErasedContextFunctionN`. With the new encoding we do not have an equivalent `XYZFunctionN`.

Reply:
Unfortunately I don't think so. I've been trying to see how poly functions deal with it, but it seems like poly functions just make scalac crash with CC on (#16871)

Reply:
With the crash fixed I think it doesn't work with poly functions as well. I wonder if we can change the parser to recognize fat arrows and add a capturing wrapper for both poly functions and erased functions?